src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationClientFactory.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System;
+using Azure;
+using Azure.Data.AppConfiguration;
+using Azure.Core;
+using Microsoft.Extensions.Azure;
+using System.Collections.Generic;
+
+namespace Microsoft.Extensions.Configuration.AzureAppConfiguration
+{
+    internal class AzureAppConfigurationClientFactory : IAzureClientFactory<ConfigurationClient>
+    {
+        private readonly ConfigurationClientOptions _clientOptions;
+
+        private readonly TokenCredential _credential;
+
+        private readonly string _secret;
+        private readonly string _id;
+
+        public AzureAppConfigurationClientFactory(
+            string connectionString,
+            ConfigurationClientOptions clientOptions)
+        {
+            if (string.IsNullOrEmpty(connectionString))
+            {
+                throw new ArgumentNullException(nameof(connectionString));
+            }
+
+            _clientOptions = clientOptions ?? throw new ArgumentNullException(nameof(clientOptions));
+
+            _secret = ConnectionStringUtils.Parse(connectionString, ConnectionStringUtils.SecretSection);
+            _id = ConnectionStringUtils.Parse(connectionString, ConnectionStringUtils.IdSection);
+        }
+
+        public AzureAppConfigurationClientFactory(
+            TokenCredential credential,
+            ConfigurationClientOptions clientOptions)
+        {
+            _credential = credential ?? throw new ArgumentNullException(nameof(credential));
+            _clientOptions = clientOptions ?? throw new ArgumentNullException(nameof(clientOptions));
+        }
+
+        public ConfigurationClient CreateClient(string endpoint)
```

Comment:
May I know why we choose to use endpoint here?

![image](https://github.com/user-attachments/assets/8d5f99c4-f877-4ece-9242-6a5cfb04a706)

According to the api and [example](https://learn.microsoft.com/en-us/dotnet/api/overview/azure/microsoft.extensions.azure-readme?view=azure-dotnet#create-named-instances)

I think the parameter of CreateClient should be a logical name. But our current implementation will rely on this endpoint to create a real app config client.

Besides, endpoint is included in the connection string. If user creates AzureAppConfigurationClientFactory with connectionstring, then this information will be redundant.

How about change this constructor to

```public AzureAppConfigurationClientFactory(
            Url endpoint
            TokenCredential credential,
            ConfigurationClientOptions clientOptions)
```

This will be consistent with our existing [connect](https://github.com/Azure/AppConfiguration-DotnetProvider/blob/main/src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationOptions.cs#L309) api.

Reply:
I think you brought up a good point about the current usage of `IAzureClientFactory`, and now I don't think it makes sense to go against the pattern for our scenario. It would probably make the most sense to create a new interface for a factory with 2 `CreateClient` APIs, one that accepts a `Uri` endpoint and one for a `string` connection string. 

@zhenlan @avanigupta @jimmyca15

Reply:
@zhiyuanliang-ms: I agree with you that it's weird to use the endpoint as the logical name. But it's the only way to solve the problem with `IAzureClientFactory`.

> endpoint is included in the connection string

@zhiyuanliang-ms: the idea was that from one replica's connection string, we can create multiple replica clients since they all share the same secrets. In that regard, `endpoint` is not redundant since there might be multiple endpoints that map to multiple clients.

That being said, I think having a custom interface for the factory instead of `IAzureClientFactory` gives us more flexibility, and we can follow the `Connect` pattern easily.

--> We will go with a custom factory interface if there are no objections.

Reply:
I just pushed a new change, the factory stores all the connection strings provided by the customer.

When trying to create a client from an endpoint given provided connection string, we try to find the one that matches the endpoint. Otherwise, we default to the first connection string (replica auto discovery scenario).